src/init.cpp

```diff
@@ -457,12 +457,13 @@ void SetupServerArgs(NodeContext& node)
     argsman.AddArg("-maxtimeadjustment", strprintf("Maximum allowed median peer time offset adjustment. Local perspective of time may be influenced by peers forward or backward by this amount. (default: %u seconds)", DEFAULT_MAX_TIME_ADJUSTMENT), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-maxuploadtarget=<n>", strprintf("Tries to keep outbound traffic under the given target (in MiB per 24h). Limit does not apply to peers with 'download' permission. 0 = no limit (default: %d)", DEFAULT_MAX_UPLOAD_TARGET), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-onion=<ip:port>", "Use separate SOCKS5 proxy to reach peers via Tor onion services, set -noonion to disable (default: -proxy)", ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
+    argsman.AddArg("-i2pproxy=<ip:port>", "SOCKS5 proxy to reach I2P peers (default: none, I2P peers are not reachable)", ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
```

Comment:
I wonder if `-proxy=<ip:port>=i2p` would make more sense

Reply:
Hmm, maybe yes. Anyway, I am working on a support for the [I2P SAM](https://geti2p.net/en/docs/api/samv3) protocol which would be used to accept incoming connections - it allows us to know that it is an I2P incoming connection and also to know the peer's I2P address. That will be `-i2psam=<addr:port>`. Once we have that we can use it also for making outgoing connections, so `-i2pproxy=` will not be needed.